plugins/build.gradle

```diff
@@ -194,10 +169,10 @@ project.parent.tasks.getByName("assemble").dependsOn << assemble
  * Merge and publish the plugins index file
  */
 task publishIndex( type: GithubRepositoryPublisher ) {
-    indexUrl = 'https://github.com/nextflow-io/plugins/main/plugins.json'
+    indexUrl = System.getenv('PLUGINS_INDEX_JSON') ?: 'https://github.com/nextflow-io/plugins/main/plugins.json'
     repos = allPlugins()
-    owner = github_organization
-    githubUser = github_username
-    githubEmail = github_commit_email
-    githubToken = github_access_token
+    owner = System.getenv('GH_ORG') ?: 'nextflow-io'
+    githubUser = System.getenv('GH_USER') ?: project.findProperty('github_username')
+    githubEmail = System.getenv('GH_USER_EMAIL') ?: project.findProperty('github_commit_email')
+    githubToken = System.getenv('GH_TOKEN') ?: project.findProperty('github_access_token')
```

Comment:
Was thinking how to avoid the proliferation of usage of env variables in the Gradle build. What is the `main.sh` write a temporary `gradle.properties` file mapping the variables to the corresponding gradle properties ?

Reply:
I've had a go at using a temporary `gradle.properties` but I don't think it's a good idea. It seems like it would be complex and fragile for very minimal benefit.

Ideally:
- running in CI, it should use the values from github (ie the env vars)
- running locally, any values from `$GRADLE_USER_HOME/gradle.properties` should take priority
- we don't want risk accidentally committing secrets to the repo
- we should try to keep things relatively simple to understand

One option would be to write the env vars into a `gradle.properties` in the project root. This would make them available to gradle, but anything in `$GRADLE_USER_HOME/gradle.properties` would still take priority thanks to gradle's built-in precedence rules. To make this safe, we'd need to add `gradle.properties` to `.gitignore`. But there's alread a committed `gradle.properties` in the project so we'd have to modify it and then somehow guarantee that it would also remove any secrets from it (including in failure scenarios) to avoid accidentally committing the changes - which is much too risky.

Another option would be to write to properties file with a different name (eg `.gradle.release.properties`) which we could explicitly `.gitignore` and load into the gradle build if detected. The problem with this is I think those properties would then override anything from `$GRADLE_USER_HOME` so we'd have to add logic to only load properties which don't already exist, or only create the properties file if running in CI - at which point it seems like a lot of unecessary complexity.

I think we should also take care to try to keep `release/main.sh` as just an orchestration script and ensure that the sub-scripts can be run independently rather than assuming they will always run as part of a main script. This will be important for testing/debugging, fixing release errors, and for future flexibility.

Reply:
I do think it's worth switching the priority order in the gradle scripts though, so that it looks for a gradle property first and then falls back to an env var:

```groovy
githubUser = project.findProperty('github_username') ?: System.getenv('GH_USER')
```

This way any values in `$GRADLE_USER_HOME/gradle.properties` will take precedence.

Reply:
> One option would be to write the env vars into a gradle.properties in the project root. 

How? or do you mean using only gradle properties?

Reply:
May be we should reverse the logic and keep all params (except secrets) in the project `gradle.properties`. 

Those could be "imported" in the shell script with basic helper, e.g. (courtesy chatgpt) 

```
# Read the property file line by line
while IFS="=" read -r key value; do
  # Skip empty lines and lines starting with # (comments)
  [[ -z "$key" || "$key" == \#* ]] && continue

  # Convert the key: replace '.' with '_' and convert to uppercase
  env_var_name=$(echo "$key" | tr '.' '_' | tr '[:lower:]' '[:upper:]')

  # Export the variable as an environment variable
  export "$env_var_name=$value"
  echo "Exported: $env_var_name=$value"
done < "$PROPERTY_FILE"
```

Secrets could be kept both in the user gradle properties or env vars. GitHub actions secrets could be easily added to project gradle.properties. 

Bit clunky but it could make the trick.